### PR TITLE
google-chrome: 124.0.6367.201 -> 125.0.6422.60

### PR DESCRIPTION
**Pull request by jnsgruk**
## Description of changes

As highlighted in [this comment](https://github.com/NixOS/nixpkgs/pull/310859#discussion_r1599877790), the update script from `google-chrome` could, in unusual cases, get an empty version - in which case the script should fail.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/go/google-chrome/package.nix [hunk 1]
```diff
@@ -146,6 +146,7 @@ in stdenv.mkDerivation (finalAttrs: {
     updateScript = writeScript "update-google-chrome.sh" ''
       #!/usr/bin/env nix-shell
       #!nix-shell -i bash -p curl jq common-updater-scripts
+      set -euo pipefail
       url="https://versionhistory.googleapis.com/v1/chrome/platforms/linux/channels/stable/versions/all/releases"
       response=$(curl --silent $url)
```
Comment (JohnRTitor) [inline pkgs/by-name/go/google-chrome/package.nix:151]: Quote `response` and `version` here.

File: pkgs/by-name/go/google-chrome/package.nix [hunk 2]
```diff
@@ -146,6 +146,7 @@ in stdenv.mkDerivation (finalAttrs: {
...
       version=$(jq ".releases[0].version" --raw-output <<< "$response")
```
Comment (JohnRTitor) [inline pkgs/by-name/go/google-chrome/package.nix:152]: Don't double wrap response and version below.

**Review by JohnRTitor — COMMENTED** [review]
Can you also update Chrome with this PR? Afaik `.207` patch has been released.

File: pkgs/by-name/go/google-chrome/package.nix [hunk 3]
```diff
@@ -146,10 +146,11 @@ in stdenv.mkDerivation (finalAttrs: {
     updateScript = writeScript "update-google-chrome.sh" ''
       #!/usr/bin/env nix-shell
       #!nix-shell -i bash -p curl jq common-updater-scripts
+      set -euo pipefail
       url="https://versionhistory.googleapis.com/v1/chrome/platforms/linux/channels/stable/versions/all/releases"
-      response=$(curl --silent $url)
-      version=$(jq ".releases[0].version" --raw-output <<< "$response")
-      update-source-version ${finalAttrs.pname} "$version" --ignore-same-hash
+      response="$(curl --silent $url)"
```
Comment (SuperSandro2000) [inline pkgs/by-name/go/google-chrome/package.nix:151]:
```suggestion
      response="$(curl --silent --fail $url)"
```

File: pkgs/by-name/go/google-chrome/package.nix [hunk 4]
```diff
@@ -64,7 +64,7 @@ let
 
 in stdenv.mkDerivation (finalAttrs: {
   pname = "google-chrome";
-  version = "124.0.6367.201";
+  version = "124.0.6367.207";
```
Comment (JohnRTitor) [inline pkgs/by-name/go/google-chrome/package.nix:67]: Did you forget to update the hash?

**Review by JohnRTitor — CHANGES_REQUESTED** [review]

File: pkgs/by-name/go/google-chrome/package.nix (hunk 3 above)
Comment (jnsgruk) [reply pkgs/by-name/go/google-chrome/package.nix:151]: Ah I see -- yes sorry I missed that nuance in your last response. Fixing

File: pkgs/by-name/go/google-chrome/package.nix (hunk 4 above)
Comment (jnsgruk) [reply pkgs/by-name/go/google-chrome/package.nix:67]: :man_facepalming:

**Comment by jnsgruk** [comment]
Clearly having a slow day. Should be good now.

**Review by JohnRTitor — APPROVED** [review]
Looked at the changes and looks good to me.

Successfully ran the update using
```bash
env NIXPKGS_ALLOW_UNFREE=1 nix run github:jnsgruk/nixpkgs/chrome-update-script-safety#google-chrome --impure
```

Note that after this is merged, you'd need to backport this PR along with the d95e9bf3a61395f00b18379d9fb6388d3e6645f9 commit, as it can't be backported automatically.

**Comment by jnsgruk** [comment]
(Removed backport label, as I'll do the above as suggested by @JohnRTitor)

**Review by JohnRTitor — CHANGES_REQUESTED** [review]
PR title should be updated.

**Pull request merged**